frontend/components/Home/Dashboard/Cards/CourseCard.tsx

```diff
@@ -1,19 +1,30 @@
 import React, { useState } from "react";
-import { Segment, Header } from "semantic-ui-react";
+import { Segment, Header, Dropdown, Icon } from "semantic-ui-react";
 import Link from "next/link";
-import { Course } from "../../../../types";
+import { UserMembership, Kind } from "../../../../types";
 
 interface CourseCardProps {
-    course: Course;
+    membership: UserMembership;
+    setOpenLeave?: (open: boolean) => void;
```

Comment:
More of a side comment - the types as is make it so that it is possible for a student course card to not have setOpenLeave and setLeaveMembership, which is undesirable since we want to enforce that those functions exist if the membership is a student. 

However, I'm actually not sure how to enforce this with types cleanly. If you happen to want to attempt a nice type exercise feel free to think about how this can be fixed, but it's not the most important thing in the world. If you want to leave it as is and get this merged go for it!